cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -319,83 +335,87 @@ func createPacketManagerRest(configReader io.Reader, discoverOpts cloudprovider.
 	return &manager, nil
 }
 
-func (mgr *packetManagerRest) listPacketDevices() (*Devices, error) {
-	var jsonStr = []byte(``)
-	packetAuthToken := os.Getenv("PACKET_AUTH_TOKEN")
-	url := mgr.getNodePoolDefinition("default").baseURL + "/projects/" + mgr.getNodePoolDefinition("default").projectID + "/devices"
-	req, _ := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
-	req.Header.Set("X-Auth-Token", packetAuthToken)
+func (mgr *packetManagerRest) request(ctx context.Context, method, url string, jsonData []byte) ([]byte, error) {
```

Comment:
I refactored this a bit to move the common request handling logic to a single method, previously there was a lot of repetition of the request handling logic (especially around error handling)